aws-cpp-sdk-transfer/source/transfer/TransferManager.cpp

```diff
@@ -268,7 +268,12 @@ namespace Aws
                     PartPointer partPtr = partsIter->second;
                     Aws::S3::Model::UploadPartRequest uploadPartRequest = m_transferConfig.uploadPartTemplate;
                     uploadPartRequest.SetContinueRequestHandler([handle](const Aws::Http::HttpRequest*) { return handle->ShouldContinue(); });
-                    uploadPartRequest.SetDataSentEventHandler([handle, partPtr](const Aws::Http::HttpRequest*, long long amount){ partPtr->OnDataTransferred(amount, handle); });
+                    auto dataEventHandler = [this, handle, partPtr](const Aws::Http::HttpRequest*, long long amount)
+                    {
```

Comment:
nitpick: can you avoid introducing the local variable `dataEventHandler`? just add the missing function call to the existing body.

Thanks for the patch.